types/ramda/index.d.ts

```diff
@@ -1172,8 +1177,8 @@ declare namespace R {
         /**
          * Returns a partial copy of an object omitting the keys specified.
          */
-        omit<T>(names: ReadonlyArray<string>, obj: T): T;
-        omit(names: ReadonlyArray<string>): <T>(obj: T) => T;
+        omit<T, K extends keyof T>(names: ReadonlyArray<K>, obj: T): Omit<T, K>;
+        omit<K extends keyof { [index: string]: any }>(names: ReadonlyArray<K>): <T>(obj: T) => Omit<T, K>;
```

Comment:
Isn't `keyof { [index: string]: any }` just `string`?

Reply:
yes. Good call.

Additionally, after reviewing the documentation here: https://blogs.msdn.microsoft.com/typescript/2018/03/27/announcing-typescript-2-8/

It does not appear that the Exclude/Extract types will have the behavior desired here.

Reply:
Actually this won't work. When you change it to string it ruins the conditional typing that constrains the structure of the return value to which we bind the projection. The trouble is not that we want all strings, but that we want the set of indices present in the specific type the pick and omit functions are used on. If we don't do this, we can't guarantee how the projections behave when this library is used.

Unfortunately the tradeoff here is the readability of the type definitions vs the accuracy of the types that these functions produce. Since these definitions serve to aid the users I think that having a more complicated types that exclude more incorrect programs while including correct ones should be the tradeoff we make.

The desired change you requested would appear like the following:
```ts
omit(names: ReadonlyArray<string>): <T>(obj: T) => Omit<T, ???>
```
As you can see, we can't actually describe the set of keys that should be missing in the return type. By making it the way it is, we can preserve not just the fact that the keys being excluded are strings, but are in fact the exact strings that are reflected in the array. This only works because the strings are themselves types in typescript.

Reply:
Upon further inspection it appears that I misunderstood what you were asking. We can simply change the following:
```ts
K extends keyof { [index: string]: any }
```
to
```ts
K extends string
```

Reply:
Yeah sorry for the unclear message earlier, you got exactly what I mean in the end